js/repeater.js

```diff
@@ -736,40 +737,44 @@
 					addItem(this.$canvas, addBefore);
 				}
 
-				var $container = this.$canvas.find('[data-container="true"]:last');
-				$container = ($container.length > 0) ? $container : this.$canvas;
+				var $dataContainer = this.$canvas.find('[data-container="true"]:last');
+				var $container = ($dataContainer.length > 0) ? $dataContainer : this.$canvas;
 
+				// It appears that the following code would theoretically allow you to pass a deeply
+				// nested value to "repeat on" to be added to the repeater.
+				// eg. `data.foo.bar.items`
 				if (viewTypeObj.renderItem) {
 					var subset;
-					var repeat = viewTypeObj.repeat || 'data.items';
-					repeat = repeat.split('.');
-					if (repeat[0] === 'data' || repeat[0] === 'this') {
-						subset = (repeat[0] === 'this') ? this : data;
-						repeat.shift();
-					} else {
-						repeat = [];
-						subset = [];
-						logWarn('WARNING: Repeater plugin "repeat" value must start with either "data" or "this"');
-					}
-
-					for (var repeatI = 0, repeatL = repeat.length; repeatI < repeatL; repeatI++) {
-						if (subset[repeat[repeatI]] !== undefined) {
-							subset = subset[repeat[repeatI]];
-						} else {
-							subset = [];
-							logWarn('WARNING: Repeater unable to find property to iterate renderItem on.');
-							repeatI = repeat.length;
+					var objectAndPropsToRepeatOnString = viewTypeObj.repeat || 'data.items';
+					var objectAndPropsToRepeatOn = objectAndPropsToRepeatOnString.split('.');
+					var objectToRepeatOn = objectAndPropsToRepeatOn[0];
+
+					if (objectToRepeatOn === 'data' || objectToRepeatOn === 'this') {
+						subset = (objectToRepeatOn === 'this') ? this : data;
+
+						// Extracts subset from object chain (get `items` out of `foo.bar.items`). I think....
+						var propsToRepeatOn = objectAndPropsToRepeatOn.slice(1);
+						for (var prop = 0; prop < propsToRepeatOn.length; prop++) {
+							if (subset[propsToRepeatOn[prop]] !== undefined) {
+								subset = subset[propsToRepeatOn[prop]];
+							} else {
+								subset = [];
+								logWarn('WARNING: Repeater unable to find property to iterate renderItem on.');
+								break;
```

Comment:
TBH the `break` was in the original code. I took it out for clarity somewhere along the way, and then once the code was simplified to this point I realized it was now actually _less clear_, and so I put it back in. :P